### ISSUE TITLE
PyScript's console throws an IO Error when using input (2024.9.1)

### ISSUE THREAD

**Issue by FerdinandoPH** [pyscript/pyscript#2178]
### Checklist

- [X] I added a descriptive title
- [X] I searched for other issues and couldn't find a solution or duplication
- [X] I already searched in Google and didn't find any good information or help

### What happened?

Pyscript's terminal crashes whenever the input function is used. At first, I thought it might be related with my code, but it crashes even with the example found on [https://docs.pyscript.net/2024.9.1/user-guide/terminal/](url)
I'm using pyscript 2024.9.1, which is the latest version I'm aware of

### What browsers are you seeing the problem on? (if applicable)

Chrome

### Console info

```shell
What is your name? Traceback (most recent call last):
  File "/lib/python312.zip/_pyodide/_base.py", line 596, in eval_code_async
    await CodeRunner(
  File "/lib/python312.zip/_pyodide/_base.py", line 410, in run_async
    coroutine = eval(self.code, globals, locals)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<exec>", line 1, in <module>
OSError: [Errno 29] I/O error
```

### Additional Context
I'm running my code from Live Server, Vscode

My code:

```
<html>
    <head>
        <meta charset="UTF-8">
        <link rel="icon" href="favicon.png" type="image/gif" >
        <title>Pocha online web</title>
        <link rel="stylesheet" href="https://pyscript.net/releases/2024.9.1/core.css">
        <script type="module" src="https://pyscript.net/releases/2024.9.1/core.js"></script>
        
    </head>
    <body>
        <script type="py" terminal worker>
            name = input("What is your name? ")
            print(f"Hello, {name}")
        </script>
    </body>
</html>
```

**Comment by WebReflection** [comment]
literally just tried this without issues:

```html
<!DOCTYPE html>
<html>
    <head>
        <meta charset="UTF-8">
        <link rel="stylesheet" href="https://pyscript.net/releases/2024.9.1/core.css">
        <script type="module" src="https://pyscript.net/releases/2024.9.1/core.js"></script>
        
    </head>
    <body>
        <script type="py" terminal worker>
            name = input("What is your name? ")
            print(f"Hello, {name}")
        </script>
    </body>
</html>
```

I believe it's a classic "*wrong headers*" situation [well explained in our F.A.Q. section](https://docs.pyscript.net/2024.9.1/faq/#sharedarraybuffer). Also the very first thing you find in [pyscript.net](https://pyscript.net/) is a working terminal with `input` working as expected.

**Comment by WebReflection** [comment]
*TL;DR* `input` can work only out of *SharedArrayBuffer* primitive ... if that's not available, it cannot block the main thread waiting to assign that `input` value typed somewhere else. If you check your devtools I am sure you'll find more details about this error and I am also confident it's just about not having *SharedarrayBuffer* enabled.
